### PR TITLE
[TS-6]: Build launch screen animation

### PR DESCRIPTION
**Pull request by MuhammadARehman**
##jira
[TS-6](https://shop-go.atlassian.net/browse/TS-6)

Description: 
- Cart should move left to right

How to test it:
- Change the storyboard name in infoplist
- Reopen the app

screenshots&gifs:
Screen Shot 2022-09-20 at 11 33 39 PM
https://user-images.githubusercontent.com/113715582/191425791-94f6af74-4f25-4dbe-8754-a64a35b47e7c.mov

### CODE REVIEW

File: ShopGo/LaunchScreenAnimationViewController.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    @IBOutlet weak var logoImage: UIImageView!
+    @IBOutlet weak var shopGoLabel: UILabel!
+    
+    override func viewDidLoad() {
+           super.viewDidLoad()
+           shopGoLabel.isHidden = true
+           DispatchQueue.main.asyncAfter(deadline: .now()+0.5, execute: {
+               self.animateForMoving()
+           })
```
Comment (haroonmaqsoodch) [inline ShopGo/LaunchScreenAnimationViewController.swift:20]: make a method and call that in viewDidLoad()

File: ShopGo/LaunchScreenAnimationViewController.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+           DispatchQueue.main.asyncAfter(deadline: .now()+1.8, execute: {
+               self.shopGoLabel.isHidden = false
+               self.animationForFading()
+           })
+           DispatchQueue.main.asyncAfter(deadline: .now()+3, execute:{
+               self.transiton()
+           })
+       }
+    func animateForMoving(){
```
Comment (haroonmaqsoodch) [inline ShopGo/LaunchScreenAnimationViewController.swift:29]: one space before the method

File: ShopGo/LaunchScreenAnimationViewController.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        let animation = CABasicAnimation()
+        animation.keyPath = "position.x"
+        animation.fromValue = logoImage.layer.position.x
+        animation.toValue = logoImage.layer.position.x + 300
+        animation.duration = 1.5
+        logoImage.layer.add(animation, forKey: "launch")
+        logoImage.layer.position = CGPoint(x: logoImage.layer.position.x + 300, y: logoImage.layer.position.y)
+    }
+    func animationForFading(){
```
Comment (haroonmaqsoodch) [inline ShopGo/LaunchScreenAnimationViewController.swift:38]: one space before the method

File: ShopGo/LaunchScreenAnimationViewController.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        let fadeIn = CABasicAnimation(keyPath: "opacity")
+        fadeIn.fromValue = 0
+        fadeIn.toValue = 1
+        fadeIn.duration = 0.5
+        shopGoLabel.layer.add(fadeIn, forKey: nil)
+    }
+    func transiton(){
```
Comment (haroonmaqsoodch) [inline ShopGo/LaunchScreenAnimationViewController.swift:45]: one space before the method

File: ShopGo/LaunchScreenAnimationViewController.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        /*
+         Use this function to naviagte to next screen
+         */
+    }
+    
+
+
```
Comment (haroonmaqsoodch) [inline ShopGo/LaunchScreenAnimationViewController.swift:52]: remove extra spaces

**Review by haroonmaqsoodch — COMMENTED** [review]
put any image and upload a gif or screen recording to show the animation

**Comment by basitalich** [comment]
Your branch name is not valid, it should be feature/yourStoryNumberAndName.

**Comment by MuhammadARehman** [comment]
> Your branch name is not valid, it should be feature/yourStoryNumberAndName.

i updated the branch name locally and updating it on github has caused it to close any suggestion on that

**Pull request closed without merging**